pkg/admission/types.go

```diff
@@ -31,3 +32,9 @@ type WantsInformerFactory interface {
 	SetInformerFactory(informers.SharedInformerFactory)
 	Validator
 }
+
+// WantsAuthorizer defines a function which sets Authorizer for admission plugins that need it.
+type WantsAuthorizer interface {
+	SetAuthorizer(authorizer.Authorizer)
+	Validator
```

Comment:
> Why does this also include the Validator interface?

We've used the `Wants` pattern in OpenShift and one of the issues we've had is a lack of validation methods that catching initialization problems.  This forces every implementor to at least pretend to consider the need to validate the input.